data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -0,0 +1,222 @@
+        createAndAddRemedialActions();
+    }
+
+    private void createAndAddRemedialActions() {
+        Map<String, Set<PropertyBag>> linkedTopologyActions = CsaProfileCracUtils.getMappedPropertyBagsSet(topologyActionsPropertyBags, CsaProfileConstants.GRID_STATE_ALTERATION_REMEDIAL_ACTION);
+        Map<String, Set<PropertyBag>> linkedContingencyWithRAs = CsaProfileCracUtils.getMappedPropertyBagsSet(contingencyWithRemedialActionsPropertyBags, CsaProfileConstants.GRID_STATE_ALTERATION_REMEDIAL_ACTION);
+
+        for (PropertyBag parentRemedialActionPropertyBag : gridStateAlterationRemedialActionPropertyBags) {
+            String remedialActionId = parentRemedialActionPropertyBag.get(CsaProfileConstants.MRID);
+
+            try {
+                if (!linkedTopologyActions.containsKey(remedialActionId)) {
+                    csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "Remedial Action: " + remedialActionId + " will not be imported because there is no topology actions linked to that RA"));
+                    continue;
+                }
+
+                checkRemedialActionCanBeImported(parentRemedialActionPropertyBag);
+
+                String nativeRaName = parentRemedialActionPropertyBag.get(CsaProfileConstants.REMEDIAL_ACTION_NAME);
+                String tsoName = parentRemedialActionPropertyBag.get(CsaProfileConstants.TSO);
+                Optional<String> targetRemedialActionNameOpt = CsaProfileCracUtils.createRemedialActionName(nativeRaName, tsoName);
+                Optional<Integer> speedOpt = getSpeedOpt(parentRemedialActionPropertyBag.get(CsaProfileConstants.TIME_TO_IMPLEMENT));
+
+                NetworkActionAdder networkActionAdder = crac.newNetworkAction().withId(remedialActionId);
+                targetRemedialActionNameOpt.ifPresent(networkActionAdder::withName);
+                if (tsoName != null) {
+                    networkActionAdder.withOperator(tsoName.substring(tsoName.lastIndexOf("/") + 1));
+                }
+
+                speedOpt.ifPresent(networkActionAdder::withSpeed);
+
+                List<String> elementaryActions = new ArrayList<>();
+                for (PropertyBag topologyActionPropertyBag : linkedTopologyActions.get(remedialActionId)) {
+                    addElementaryActions(networkActionAdder, topologyActionPropertyBag, remedialActionId, elementaryActions);
+                }
+
+                if (elementaryActions.isEmpty()) {
+                    csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "Remedial Action: " + remedialActionId + " will not be imported because none of the topology actions linked to that RA has a Switch that matches a switch in the network model"));
+                    continue;
+                }
```

Comment:
This can only happen if there were initially no topologyActions linked to the RA, otherwise you would have gotten a FaraoImportException from addElementaryActions. Also, since you are already catchin the exception, you can simplify the code:
```suggestion
                if (linkedTopologyActions.get(remedialActionId).isEmpty()) {
                    throw new FaraoImportException(ImportStatus.INCOMPLETE_DATA, "Remedial Action: " + remedialActionId + " will not be imported because it has no topology actions"));
                }
```